src/wovnio/wovnphp/Url.php

```diff
@@ -174,8 +183,40 @@ public static function removeLangCode($uri, $pattern, $lang_code, $lang_param_na
             case 'path':
             default:
                 // limit to one replacement
-                return preg_replace('/\/' . $lang_code . '(\/|$)/i', '/', $uri, 1);
+                $prefix = empty($site_prefix_path) ? '' : '/' . $site_prefix_path;
+                return preg_replace("@$prefix/$lang_code(/|$)@i", "$prefix/", $uri, 1);
                 break;
         }
     }
+
+    public static function matchSitePrefixPath($uri, $site_prefix_path)
+    {
+        return !!preg_match(
+            '@' .
+            '^(.*://|//)?' . // schema (optional)
+            '([^/?]*)?' . // host (optional)
+            "(/$site_prefix_path)" . // site prefix path
+            '(/|\?|#|$)' . // path, query, hash or end-of-string
+            '@',
+            $uri
+        );
+    }
+
+    private static function addPathLangCode($no_lang_url, $lang, $site_prefix_path = '')
+    {
+        if (empty($lang)) {
+            return $no_lang_url;
+        }
+        $prefix = $site_prefix_path ? '/' . $site_prefix_path : '';
+        return preg_replace(
+            '@' .
```

Comment:
any way to re-use this regex with the one in `matchSitePrefixPath`?

Reply:
Done